File: test/CodeGen/t5-autogen.py
# ----- Autogen kernel by Phaeton -----
import numpy as np

a = np.random.rand(3, 4)
b = np.random.rand(4, 3)
c = np.random.rand(3, 3)
t0 = np.tensordot(a, b, axes=[[1], [0]])
c = t0

File: test/CodeGen/t5-autogen.cpp
// ----- Autogen kernel by Phaeton -----

void omp_func(
float *a,
float *b)
{
  float c[9];
  for (unsigned i0 = 0; i0 < 3; ++i0) {
    for (unsigned i1 = 0; i1 < 3; ++i1) {
      c[(i1 + 3*(i0))] = 0.0;
      for (unsigned i2_contr = 0; i2_contr < 4; ++i2_contr) {
        c[(i1 + 3*(i0))] += a[(i2_contr + 4*(i0))] * b[(i1 + 3*(i2_contr))];
      }
    }
  }
}
